lib/GameConstants.cpp

```diff
@@ -215,6 +216,31 @@ std::string FactionID::encode(const si32 index)
 	return VLC->factions()->getByIndex(index)->getJsonKey();
 }
 
+std::string FactionID::scope()

```

Comment:
Perhaps rename to entityType / entityClass? Or anything else that does not contains "scope" in its name?

We normally use word "scope" for defining mod that generated this object/json field. Using "scope" for similar purpose might lead to mistakes.